[WGSL] Remove XCTest-based unit tests

djg opened the pull request:
#### 2b202352430fbe90c22f8dca2b6285aaae048dbb
<pre>
[WGSL] Remove XCTest-based unit tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=245714">https://bugs.webkit.org/show_bug.cgi?id=245714</a>
rdar://problem/100439888

Reviewed by Myles Maxfield.

Move the last set of tests for constant literals to gtest and remove
XCTest-based tests in WebGPU project.

* Source/WebGPU/Configurations/WGSLUnitTests.xcconfig: Removed.
* Source/WebGPU/WGSLUnitTests/WGSLLexerTests.mm: Removed.
* Source/WebGPU/WGSLUnitTests/WGSLParserTests.mm: Removed.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WGSL/ConstLiteralTests.cpp: Renamed from Source/WebGPU/WGSLUnitTests/Parser/ConstLiteralTests.mm.
(parseLCharPrimaryExpression):
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/254949@main">https://commits.webkit.org/254949@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b3a08b00dbe1481cacaaa801e70053c8ad19c69

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90767 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35346 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21345 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100068 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/158578 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/94776 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33846 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28945 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83114 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/96452 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96422 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/26947 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77579 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26764 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81714 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81465 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69795 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/34934 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15502 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32739 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16482 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3453 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36512 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/39428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38436 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35571 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

djg commented on Tools/TestWebKitAPI/Tests/WGSL/ConstLiteralTests.cpp:
```diff
@@ -43,38 +42,36 @@
-- (void)setUp {
-    [super setUp];
-    [self setContinueAfterFailure:NO];
+TEST(WGSLConstLiteralTests, DeleteBeforeLanding)
```
To test that TestWGSL is running on EWS, this should fail. Will remove before merging.

litherum commented on Tools/TestWebKitAPI/Tests/WGSL/ConstLiteralTests.cpp:
```diff
@@ -132,21 +131,22 @@ - (void)testAbstractFloatLiteralDec {
-    XCTSkip(@"Hexadecimal floats not implemented yet");
-
+// FIXME: Implement hexadecimal float parsing.
+#if 0
```
We don't generally commit dead code.

djg replied:
Removed.

litherum commented on Tools/TestWebKitAPI/Tests/WGSL/ConstLiteralTests.cpp:
```diff
@@ -43,38 +42,36 @@
 };
 
-@interface ConstLiteralTests : XCTestCase
+namespace TestWebKitAPI {
```
Does TestWTFAPI use the `TestWebKitAPI` namespace? Should we make a TestWGSLAPI namespace?

djg committed:
[WGSL] Remove XCTest-based unit tests
https://bugs.webkit.org/show_bug.cgi?id=245714
rdar://problem/100439888

Reviewed by Myles Maxfield.

Move the last set of tests for constant literals to gtest and remove
XCTest-based tests in WebGPU project.

* Source/WebGPU/Configurations/WGSLUnitTests.xcconfig: Removed.
* Source/WebGPU/WGSLUnitTests/WGSLLexerTests.mm: Removed.
* Source/WebGPU/WGSLUnitTests/WGSLParserTests.mm: Removed.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WGSL/ConstLiteralTests.cpp: Renamed from Source/WebGPU/WGSLUnitTests/Parser/ConstLiteralTests.mm.
(parseLCharPrimaryExpression):
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/254949@main
